### PR TITLE
Search related UI

### PR DESCRIPTION
**Pull request by bun919tw**
Overview         --\
                            |--> Search Activity --> Search Fragment
CardCollection --/                                   (extends from titled fragment)

### CODE REVIEW

**Comment by bun919tw** [comment]
我畫的精美的圖竟然顯示不出來
要怎麼在 github conversation 上顯示空白啊

File: PointTreeUser/app/src/main/res/drawable/search_bar_style.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+    <item>
+        <shape>
+            <solid android:color="#FFFFFF" />
+
+            <stroke
+                android:width="1dp"
+                android:color="@color/colorPrimary" />
+            <corners
```
Comment (StanleyDing) [inline PointTreeUser/app/src/main/res/drawable/search_bar_style.xml]: Use android:radius to set all corners.

File: PointTreeUser/app/src/main/res/drawable/search_bar_style.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+    <item>
+        <shape>
+            <solid android:color="#FFFFFF" />
```
Comment (StanleyDing) [inline PointTreeUser/app/src/main/res/drawable/search_bar_style.xml]: There is `@color/white`

**Comment by StanleyDing** [comment]
Remove file `values-v21/styles.xml` to fix white status bar.

**Comment by bun919tw** [comment]
@StanleyDing @boolafish 我改好了～請看

File: PointTreeUser/app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -18,6 +18,14 @@
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+        <activity
+            android:name=".activities.SearchActivity"
+            android:label="@string/title_activity_search"
+            android:theme="@style/AppTheme.NoActionBar">
+            <intent-filter>
+                <action android:name="android.intent.action.SEARCH" />
```
Comment (StanleyDing) [inline PointTreeUser/app/src/main/AndroidManifest.xml]: Is this used?

File: PointTreeUser/app/src/main/res/drawable/search_bar_style.xml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+            <stroke
+                android:width="1dp"
+                android:color="@color/colorPrimary" />
+
+            <corners
+                android:radius="10dp" />
+        </shape>
+    </item>
+</selector>
```
Comment (StanleyDing) [inline PointTreeUser/app/src/main/res/drawable/search_bar_style.xml]: Missing a newline.

File: PointTreeUser/app/src/main/res/layout/fragment_search.xml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+            android:layout_height="wrap_content"
+            android:id="@+id/search_bar"
+            style="@style/AppTheme.SearchBarActive" />
+    </LinearLayout>
+
+    <ListView
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:id="@+id/search_list"/>
+</LinearLayout>
```
Comment (StanleyDing) [inline PointTreeUser/app/src/main/res/layout/fragment_search.xml]: Missing newline.

File: PointTreeUser/app/src/main/res/values/styles.xml [hunk 6]
```diff
@@ -25,4 +25,21 @@
         <item name="android:textSize">21sp</item>
     </style>
 
+    <style name="AppTheme.SearchBarInactive">
```
Comment (StanleyDing) [inline PointTreeUser/app/src/main/res/values/styles.xml]: Use style inheritance to avoid writing duplicate items.

**Comment by bun919tw** [comment]
done

**Pull request merged**